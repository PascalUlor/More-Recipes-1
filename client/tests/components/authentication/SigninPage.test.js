/* eslint-disable */
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import { SigninPage } from '../../../components/SigninPage.jsx';


/**
 * @param { boolean } loading
 *
 * @return { * } null
 */
const setup = () => {
  const props = {
    isAuthenticated: false,
    flashMessage: {},
    addFlashMessage: jest.fn(),
    userSigninRequest: jest.fn(() => Promise.resolve()),
    history: {
      push: () => {}
    }
  };
  const store = mockStore({});
  return mount((
    <Provider store={store}>
      <Router><SigninPage {...props}/></Router>
    </Provider>
  ));
};

describe('<SigninPage/>', () => {
  const mountWrapper = setup();
  it('should render without crashing', () => {
    expect(toJson(mountWrapper)).toMatchSnapshot();
  });

  it('should call onChange() event', () => {
    mountWrapper.find('SigninForm')
      .find('TextFieldGroup').at(0).find('input')
      .simulate('change', {
        target: {
          name: 'username',
          value: 'rich',
        },
      });
    expect(mountWrapper.find('SigninPage')
      .instance().state.username).toEqual('rich');
  });

  it('should call onFocus() event', () => {
    mountWrapper.find('SigninForm')
      .find('TextFieldGroup').at(0).find('input')
      .simulate('focus');
  });

  describe('handleSubmit()', () => {
    const event = {
      preventDefault: jest.fn()
    };
    const signinPage = mountWrapper.find('SigninPage').instance();
    const submit = () => mountWrapper.find('SigninForm')
      .find('TextFieldGroup').at(0).find('input')
      .simulate('submit', event);
    it('should not work for empty username', () => {
      signinPage.setState({
        username: '',
        password: ''
      });
      submit();
      expect(signinPage.state.errors)
        .toEqual({
          username: 'Username is required',
          password: 'Password is required'
        });
      expect(signinPage.props.userSigninRequest).not.toBeCalled();
    });

    it('should work if provided form details are valid', () => {
      signinPage.setState({
        username: 'rich',
        password: '12345678'
      });
      submit();
      expect(signinPage.state.errors).toEqual({});
      expect(signinPage.props.userSigninRequest).toBeCalled();
    });
  });
});
