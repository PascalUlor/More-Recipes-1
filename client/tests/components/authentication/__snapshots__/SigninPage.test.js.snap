// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SigninPage/> should render without crashing 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <SigninPage
        addFlashMessage={[MockFunction]}
        flashMessage={Object {}}
        history={
          Object {
            "push": [Function],
          }
        }
        isAuthenticated={false}
        userSigninRequest={[MockFunction]}
      >
        <div>
          <div
            className="site-wrapper"
          >
            <div
              className="container-fluid mt-4"
              style={
                Object {
                  "height": "85.5vh",
                }
              }
            >
              <div
                className="row"
              >
                <div
                  className="col-sm-10 col-md-8 col-lg-8 offset-sm-1 offset-md-2 offset-lg-2"
                >
                  <div
                    className="row"
                  >
                    <SigninSignupImage>
                      <div
                        className="col-4 col-sm-4 col-md-4 col-lg-4"
                      >
                        <figure
                          className="figure"
                        >
                          <Link
                            replace={false}
                            to="/"
                          >
                            <a
                              href="/"
                              onClick={[Function]}
                            >
                              <img
                                alt="More Recipes Logo"
                                className="figure-img img-fluid rounded"
                                height="70"
                                src="/images/logo.png"
                                width="200"
                              />
                              <figcaption
                                className="figure-caption text-center"
                              >
                                <h3
                                  id="site-name"
                                >
                                  More Recipes
                                </h3>
                              </figcaption>
                            </a>
                          </Link>
                        </figure>
                      </div>
                    </SigninSignupImage>
                    <SigninForm
                      errors={Object {}}
                      handleChange={[Function]}
                      handleFocus={[Function]}
                      handleSubmit={[Function]}
                      isLoading={false}
                      password=""
                      showWarning={false}
                      username=""
                    >
                      <div
                        className="col-8 col-sm-8 col-md-8 col-lg-8"
                      >
                        <h2>
                          Sign In
                        </h2>
                        <p
                          className="lead"
                        >
                          Don't have a More-Recipes account?
                          <Link
                            replace={false}
                            to="/signup"
                          >
                            <a
                              href="/signup"
                              onClick={[Function]}
                            >
                              Sign Up
                            </a>
                          </Link>
                        </p>
                        <form
                          onSubmit={[Function]}
                          role="form"
                        >
                          <TextFieldGroup
                            font="fa fa-user"
                            label="Username"
                            name="username"
                            onChange={[Function]}
                            onFocus={[Function]}
                            placeholder="enter username"
                            type="text"
                            value=""
                          >
                            <div
                              className="form-group"
                            >
                              <label
                                className="page-text"
                                htmlFor="username"
                              >
                                Username
                              </label>
                              <div
                                className="input-group mb-2 mr-sm-2 mb-sm-0"
                              >
                                <div
                                  className="input-group-addon page-text"
                                >
                                  <i
                                    aria-hidden="true"
                                    className="fa fa-user"
                                  />
                                </div>
                                <input
                                  className="form-control form-control-sm"
                                  defaultValue=""
                                  id="username"
                                  name="username"
                                  onChange={[Function]}
                                  onFocus={[Function]}
                                  placeholder="enter username"
                                  type="text"
                                />
                              </div>
                            </div>
                          </TextFieldGroup>
                          <TextFieldGroup
                            font="fa fa-lock"
                            label="Password"
                            name="password"
                            onChange={[Function]}
                            onFocus={[Function]}
                            placeholder="enter password"
                            type="password"
                            value=""
                          >
                            <div
                              className="form-group"
                            >
                              <label
                                className="page-text"
                                htmlFor="password"
                              >
                                Password
                              </label>
                              <div
                                className="input-group mb-2 mr-sm-2 mb-sm-0"
                              >
                                <div
                                  className="input-group-addon page-text"
                                >
                                  <i
                                    aria-hidden="true"
                                    className="fa fa-lock"
                                  />
                                </div>
                                <input
                                  className="form-control form-control-sm"
                                  defaultValue=""
                                  id="password"
                                  name="password"
                                  onChange={[Function]}
                                  onFocus={[Function]}
                                  placeholder="enter password"
                                  type="password"
                                />
                              </div>
                            </div>
                          </TextFieldGroup>
                          <button
                            className="btn btn-outline-success"
                            disabled={false}
                            type="submit"
                          >
                            Sign In 
                          </button>
                        </form>
                      </div>
                    </SigninForm>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <Footer>
            <footer
              className="footer"
            >
              <div
                className="container"
              >
                <div
                  className="row mt-4"
                >
                  <div
                    className="col-12 col-sm-8 col-md-9 col-lg-10"
                  >
                    <small
                      className="text-white"
                      id="copywrite"
                    >
                      <span
                        className="text-warning"
                      >
                        Â©2018 
                      </span>
                      Bootcamp27, Andela Nigeria. All rights reserved.
                    </small>
                  </div>
                  <div
                    className="col-12 col-sm-4 col-md-3 col-lg-2 text-center"
                  >
                    <ul
                      className="list-inline"
                    >
                      <li
                        className="list-inline-item"
                      >
                        <a
                          href="#"
                        >
                          <i
                            aria-hidden="true"
                            className="fa fa-google-plus"
                          />
                        </a>
                      </li>
                      <li
                        className="list-inline-item"
                      >
                        <a
                          href="#"
                        >
                          <i
                            className="fa fa-github"
                          />
                        </a>
                      </li>
                      <li
                        className="list-inline-item"
                      >
                        <a
                          href="#"
                        >
                          <i
                            className="fa fa-facebook"
                          />
                        </a>
                      </li>
                      <li
                        className="list-inline-item"
                      >
                        <a
                          href="#"
                        >
                          <i
                            className="fa fa-twitter"
                          />
                        </a>
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </footer>
          </Footer>
        </div>
      </SigninPage>
    </Router>
  </BrowserRouter>
</Provider>
`;
